### PR TITLE
[OCPBUGS-3829]: Add etcd module from KCS article

### PR DESCRIPTION
**Pull request by tmalove**
[OCPBUGS-3829](https://issues.redhat.com/browse/OCPBUGS-3829)

Version(s): 4.9+

Link to docs preview (updated 3/28): http://file.rdu.redhat.com/tlove/etcd-ocpbugs-3829-tlove-new/scalability_and_performance/recommended-host-practices.html#move-etcd-different-disk_recommended-host-practices

QE review:
- [ x] QE has approved this change.

Additional information: [access.redhat.com/solutions/6973069](https://access.redhat.com/solutions/6973069)
This Jira is to add the content from this [KCS](https://github.com/openshift/openshift-docs/pull/url) article to the docs.

### CODE REVIEW

**Comment by tmalove** [comment]
@geliu2016  Will you ack this module update? Thanks!

File: modules/move-etcd-different-disk.adoc [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        name: var-lib-etcd.mount
+----
+. Apply the modified version that removes the logic for creating and syncing the device by entering the following command:
++
+[source,terminal]
+----
+$ oc replace -f etcd-mc.yml
+----
++
+The previous step prevents the nodes from rebooting.
```
Comment (geliu2016) [inline modules/move-etcd-different-disk.adoc:182]: @tmalove what means of "The previous step prevents the nodes from rebooting."? which previous step? thanks

Comment (geliu2016) [reply modules/move-etcd-different-disk.adoc:182]: I suppose it's "oc replace" step, exact?

**Comment by michaelalang** [comment]
@geliu2016 it just seems to be a mis understanding from the original KCS mapping to the Docs ...

```
$ oc replace -f etcd-mc.yml
this step is expected to not cause a reboot of the nodes
```

https://access.redhat.com/solutions/6973069

**Comment by geliu2016** [comment]
> @geliu2016 it just seems to be a mis understanding from the original KCS mapping to the Docs ...
> 
> ```
> $ oc replace -f etcd-mc.yml
> this step is expected to not cause a reboot of the nodes
> ```
> 
> https://access.redhat.com/solutions/6973069

Great to know, thank you, Michaela

**Comment by tmalove** [comment]
@geliu2016 @michaelalang  I will move that sentence (about ...prevents the nodes from...) into  Step 4, for clarity. Thanks!

File: modules/move-etcd-different-disk.adoc [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+$ oc create -f etcd-mc.yml
+machineconfig.machineconfiguration.openshift.io/98-var-lib-etcd created
+----
++
+[source,terminal]
+----
+$ oc login -u ${ADMIN} -p ${ADMINPASSWORD} ${API} [... output omitted ...] $ oc create -f etcd-mc.yml machineconfig.machineconfiguration.openshift.io/98-var-lib-etcd created
+----
++
+The nodes update and reboot. After the reboot completes, the following events occur:
```
Comment (lahinson) [inline modules/move-etcd-different-disk.adoc:134]:
```suggestion
The nodes are updated and rebooted. After the reboot is completed, the following events occur:
```
A couple of minor suggestions here. Unless the nodes update and reboot themselves, I think we need to adjust the wording to indicate that they are updated and rebooted by some other entity.

**Review by lahinson — COMMENTED** [review]
Nice work, Tami! I made a couple of very minor suggestions. Thanks for working on this one!

File: modules/move-etcd-different-disk.adoc [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+$ oc create -f etcd-mc.yml
+machineconfig.machineconfiguration.openshift.io/98-var-lib-etcd created
+----
++
+[source,terminal]
+----
+$ oc login -u ${ADMIN} -p ${ADMINPASSWORD} ${API} [... output omitted ...] $ oc create -f etcd-mc.yml machineconfig.machineconfiguration.openshift.io/98-var-lib-etcd created
+----
++
+The nodes are updated and rebooted. After the reboot is completed, the following events occur:
```
Comment (jldohmann) [inline modules/move-etcd-different-disk.adoc:137]:
```suggestion
The nodes are updated and rebooted. After the reboot completes, the following events occur:
```

File: modules/move-etcd-different-disk.adoc [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
++
+*  An XFS filesystem is created on the specified disk.
```
Comment (jldohmann) [inline modules/move-etcd-different-disk.adoc:139]:
```suggestion
*  An XFS file system is created on the specified disk.
```

**Review by jldohmann — COMMENTED** [review]
it's looking good! left some feedback below

**Comment by mjpytlak** [comment]
@tmalove For future reference, please do not remove `peer-review-done`.  The squad uses that latter to make sure the peer review is complete ahead of a merge.

**Comment by tmalove** [comment]
> @tmalove For future reference, please do not remove `peer-review-done`. The squad uses that latter to make sure the peer review is complete ahead of a merge.

Apologies @mjpytlak ...I have it in my 'smart reply'. I will remove it now.

**Comment by mjpytlak** [comment]
No worries @tmalove. However, ahead of a merge please address https://github.com/openshift/openshift-docs/pull/57592/files#r1151029235. Not sure if you intentionally left it or missed the comment.

File: modules/move-etcd-different-disk.adoc [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+:_content-type: PROCEDURE
+[id="move-etcd-different-disk_{context}"]
+= Moving etcd to a different disk
+
+You can move etcd from a shared disk to a separate disk to prevent or resolve performance issues.
+
+.Prerequisites
+
+*  The `MachineConfigPool` must match `metadata.labels[“machineconfiguration.openshift.io/role]`. This applies to a controller, worker, or a custom pool.
+*  The node's auxiliary storage device, such as `/dev/sdb`, must match the sdb. Change this reference in all places in the file.
```
[the remark this answers is not in the document]
Comment (mjpytlak) [reply modules/move-etcd-different-disk.adoc:14]: > `sdb` is the block device name (/storage/disk) presented to the VM/Instance available on the path

Thanks @mtulio for the explanation. I will get you merged up @tmalove.

**Comment by mjpytlak** [comment]
@tmalove Looks like you have a conflict in 4.9. Please resolve it and give me a shout when the manual CP is ready to be merged. In the interim, I would get the other branches merged.

**Comment by tmalove** [comment]
> @tmalove Looks like you have a conflict in 4.9. Please resolve it and give me a shout when the manual CP is ready to be merged. In the interim, I would get the other branches merged.

Will do, thanks!

**Pull request merged**